### PR TITLE
add 'user_name' to AuthUser and include 'sentBy' in the SentReferralDTO

### PR DESCRIPTION
**Pull request by tomsmyers**
## What does this pull request do?

store 'user_name' claim from auth token and include in SentReferralDTO to allow the front end to look up information about the probation practitioner who sent the referral

## What is the intent behind these changes?

satisfy the latest /sent-referrals API endpoints used by the UI

### CODE REVIEW

**Review by Gweaton — APPROVED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/controller/ReferralController.kt [hunk 1]
```diff
@@ -138,9 +138,12 @@ class ReferralController(
...
       ?: throw ServerWebInputException("no 'user_id' claim in authentication token")
 
+    val userName = authentication.token.getClaimAsString("user_name")
+      ?: throw ServerWebInputException("no 'user_name' claim in authentication token")
+
     val authSource = authentication.token.getClaimAsString("auth_source")
       ?: throw ServerWebInputException("no 'auth_source' claim in authentication token")
 
-    return AuthUser(id = userID, authSource = authSource)
+    return AuthUser(id = userID, authSource = authSource, userName = userName)
```
Comment (sldblog) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/controller/ReferralController.kt:147]: _nitpick_
`userName` and `username` are used throughout the PR -- should we pick one style for consistency?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/dto/AuthUserDTO.kt [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.dto
+
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.AuthUser
+
+class AuthUserDTO(
```
Comment (sldblog) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/dto/AuthUserDTO.kt:5]: _nitpick_
Would we benefit from `id` here?

**Review by sldblog — APPROVED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/controller/ReferralController.kt (hunk 1 above)
Comment (tomsmyers) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/controller/ReferralController.kt:147]: this is a good spot. george raised something similar on a backend PR. i'll update this.

**Review by rgforsyth — APPROVED** [review]

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/dto/AuthUserDTO.kt (hunk 2 above)
Comment (tomsmyers) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/dto/AuthUserDTO.kt:5]: it's currently not used in the UI, we can add it easily later if required.

Comment (sldblog) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/dto/AuthUserDTO.kt:5]: Thinking out loud -- it feels weird to expose username/auth source as they're both used for identification and not the ID. Not a big deal :)

**Pull request merged**